### DISCUSSION TITLE
Can we use JSP code in header_include file?

### DISCUSSION THREAD

**Discussion by jkjha** [oracle/opengrok#4086] [Q&A]
I am trying to add username header at the top of my opengrok instance and for that, I am thinking of using header_include configuration mentioned here: https://github.com/oracle/opengrok/wiki/Webapp-configuration

```
<%@ page language="java" isELIgnored="false" import = "java.io.*,java.util.*,java.net.*" %>

<%
               Enumeration headerNames = request.getHeaderNames();
               while(headerNames.hasMoreElements()) {
                  String paramName = (String)headerNames.nextElement();
                  out.print("<tr><td>" + paramName + "</td>\n");
                  String paramValue = request.getHeader(paramName);
                  out.println("<td> " + paramValue + "</td></tr>\n");
               }
 %>
```

But it seems JSP code is not supported in header_include? This gets printed as HTML on UI:

<img width="1665" alt="image" src="https://user-images.githubusercontent.com/4188259/199266853-35f6c16d-95c9-47cf-b038-3d06dc72956c.png">

Can someone please help.

**Comment by vladak** [comment]
The includes accept HTML data only. See https://github.com/oracle/opengrok/blob/8a7aa08e2f11ac1acfba209e18a1444600fe18df/opengrok-web/src/main/webapp/httpheader.jspf#L109

**Comment by jkjha** [comment]
Thanks @vladak  for the help. One quick question...
so, only way to get the username or any custom data example (request.getHeader('x-email')) is editing `pageheader.jspf` file (by creating our custom docker image)? I am trying to avoid creating/ editing custom image for easy upgrade in future.

**Reply by vladak** [reply]
Yep. I'd be very hesitant to allow JSP code inserts. One can cause enough harm just with HTML/JS code.
